tests/integration_tests/performance/test_snapshot_perf.py

```diff
@@ -0,0 +1,250 @@
+                              context.disk.name()))
+
+    # Create a rw copy artifact.
+    rw_disk = context.disk.copy()
+    # Get ssh key from read-only artifact.
+    ssh_key = context.disk.ssh_key()
+
+    # Measure a burst of snapshot create calls.
+    for i in range(SAMPLE_COUNT):
+        # Create a fresh microvm from aftifacts.
+        vm = vm_builder.build(kernel=context.kernel,
+                              disks=[rw_disk],
+                              ssh_key=ssh_key,
+                              config=context.microvm,
+                              enable_diff_snapshots=enable_diff_snapshots)
+
+        # Configure metrics system.
+        metrics_fifo_path = os.path.join(vm.path, 'metrics_fifo')
+        metrics_fifo = log_tools.Fifo(metrics_fifo_path)
+
+        response = vm.metrics.put(
+            metrics_path=vm.create_jailed_resource(metrics_fifo.path)
+        )
+        assert vm.api_session.is_status_no_content(response.status_code)
+
+        vm.start()
+
+        # Create a snapshot builder from a microvm.
+        snapshot_builder = SnapshotBuilder(vm)
+        snapshot_builder.create([rw_disk],
+                                ssh_key,
+                                snapshot_type)
+        metrics = vm.flush_metrics(metrics_fifo)
+
+        if snapshot_type == SnapshotType.FULL:
+            value = metrics['latencies_us']['full_create_snapshot'] / 1000
+            baseline = CREATE_LATENCIES[context.microvm.name()]['FULL']
+        else:
+            value = metrics['latencies_us']['diff_create_snapshot'] / 1000
+            baseline = CREATE_LATENCIES[context.microvm.name()]['DIFF']
+
+        assert baseline > value, "CreateSnapshot latency degraded."
+
+        logger.info("Latency {}/5: {} ms".format(i + 1, value))
+        vm.kill()
+
+
+def _test_snapshot_resume_latency(context):
+    logger = context.custom['logger']
+    vm_builder = context.custom['builder']
+    snapshot_type = context.custom['snapshot_type']
+    enable_diff_snapshots = snapshot_type == SnapshotType.DIFF
+
+    logger.info("""Measuring snapshot resume({}) latency for microvm: \"{}\",
+kernel {}, disk {} """.format(snapshot_type,
+                              context.microvm.name(),
+                              context.kernel.name(),
+                              context.disk.name()))
+
+    # Create a rw copy artifact.
+    rw_disk = context.disk.copy()
+    # Get ssh key from read-only artifact.
+    ssh_key = context.disk.ssh_key()
+    # Create a fresh microvm from aftifacts.
+    basevm = vm_builder.build(kernel=context.kernel,
+                              disks=[rw_disk],
+                              ssh_key=ssh_key,
+                              config=context.microvm,
+                              enable_diff_snapshots=enable_diff_snapshots)
+
+    network_config = net_tools.UniqueIPv4Generator.instance()
+    _, host_ip, guest_ip = basevm.ssh_network_config(network_config,
```

Comment:
Could we use pass the parameters by name? Passing `1` is a bit confusing for someone who doesn't know the parameter order.

Reply:
This is the old framework. I would not invest in fixing the tech debt in this PR. I think we still need to rewrite this in the new framework.

Reply:
done